modules/ipc/ipcProviderBackend.js

```diff
@@ -1,670 +1,400 @@
-        this.syncEvents = {};
-        this.asyncEvents = {};
-
-
-        this.sender = event.sender;
-        this.id = event.sender.getId();
+     * Handle IPC call to destroy a connection.
+     */
+    _destroyConnection (event) {
+        const ownerId = event.sender.getId();
 
-        this.ipcSocket.setEncoding('utf8');
-        this.ipcSocket.setTimeout(0); // disable
-        // this.ipcSocket.setKeepAlive(true, 1000 * 10);
-        // this.ipcSocket.setNoDelay(false);
+        return Q.try(() => {
+            if (this._connections[ownerId]) {
+                log.debug('Destroy socket connection', ownerId);
 
-        // setup socket
-        this.connect(event);
-        this.setupSocket();
+                this._connections[ownerId].owner.send('ipcProvider-setWritable', false);
 
-        return this;
-    };
+                return this._connections[ownerId].socket.destroy().finally(() => {
+                    delete this._connections[ownerId];
+                });
+            }            
+        });
+    }
 
 
     /**
-    Connects to a socket
-
-
-    @param {Object} event     If the event param is present it assumes its a sync request and will return the writable property, using "event.returnValue"
-    @method connect
-    */
-    GethConnection.prototype.connect = function(event){
-        var _this = this,
-            timeoutId,
-            successEventFunc,
-            errorEventFunc;
-
-        if(!this.ipcSocket.writable) {
-
-            // log.info('IPCSOCKET '+ this.id +' CONNECTING..');
-
-            this.ipcSocket = this.ipcSocket.connect({path: this.path});
-
-            // make sure to set the right writeable
-            successEventFunc = function(){
-                if(event && timeoutId) {
-                    clearTimeout(timeoutId);
-                    event.returnValue = true;
-                }
-                
-                _this.ipcSocket.removeListener('error', errorEventFunc);
-            };
-            this.ipcSocket.once('connect', successEventFunc);
-
-            errorEventFunc = function(){
-                if(event && timeoutId) {
-                    clearTimeout(timeoutId);
-                    event.returnValue = false;
-                }
-             
-                _this.ipcSocket.removeListener('connect', successEventFunc);
-            };
-            this.ipcSocket.once('error', errorEventFunc);
-
-
-            // return if it takes to long
-            if(event) {
-                timeoutId = setTimeout(function(){
-                    event.returnValue = _this.ipcSocket.writable;
-                    timeoutId = null;
-                }, 500);
-            }
-        
-        } else if(event) {
-            event.returnValue = true;
+     * Handler for when Ethereum node state changes.
+     *
+     * Auto-reconnect sockets when ethereum node state changes
+     *
+     * @param {String} state The new state.
+     */
+    _onNodeStateChanged (state) {
```

Comment:
Thats problematic, as pub/sub connections won't be informed  and those are socket bound notifications.

What i do in the frontend is, reloading the dapps, which anyway creates a new socket